src/main/java/com/sprint/mission/discodeit/repository/MessageRepository.java

```diff
@@ -1,16 +1,35 @@
 package com.sprint.mission.discodeit.repository;
 
 import com.sprint.mission.discodeit.entity.Message;
-
 import java.util.List;
-import java.util.Optional;
 import java.util.UUID;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.EntityGraph;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface MessageRepository extends JpaRepository<Message, UUID> {
+
+  @EntityGraph(attributePaths = {
+      "author",
+      "author.profile",
+      "author.userStatus",
+      "attachments"
+  })
+  Page<Message> findAllByChannel_Id(@Param("channelId") UUID channelId, Pageable pageable);
```

Comment:
```suggestion
  Page<Message> findAllByChannelId(@Param("channelId") UUID channelId, Pageable pageable);
```
언더바가 안 들어가도 정상 동작할 것으로 보입니다!

Reply:
추가로, OneToMany 관계인 attachments를 함께 조회한다면 페이징이 정상동작하지 않을 수 있어요.
왜냐하면 아래처럼 조회될 수 있기 때문이죠.

message_id | message_content | attachment_id
-- | -- | --
M1 | Hello | A1
M1 | Hello | A2
M2 | Hi | B1
M2 | Hi | B2

아래 링크 보면 도움이 많이 될거에요!

https://tecoble.techcourse.co.kr/post/2020-10-21-jpa-fetch-join-paging/
https://jojoldu.tistory.com/737